api/src/data/migrations/019.create-actions.ts

```diff
@@ -0,0 +1,30 @@
+import * as knex from "knex";
+
+export async function up(knex: knex.Knex) {
+    await knex.schema.createTable("actions", function(table) {
+        table.increments("id").primary().notNullable();
+        table.integer("hazard_id").notNullable();
+        table.integer("creator_employee_id").nullable();
+        table.integer("creator_role_id").nullable();
+        table.integer("actor_employee_id").nullable();
+        table.integer("actor_role_id").nullable();
+        table.datetime("created_at").notNullable().defaultTo(knex.fn.now());
+        table.datetime("modified_at").nullable().defaultTo(knex.fn.now());
+        table.datetime("due_date").nullable();
+        table.string("description", 4096).notNullable();
+        table.string("action_type", 8).notNullable();
+        table.string("sensitivity", 8).nullable();
+        table.string("status", 8).nullable();
+
+        table.foreign("hazard_id").references("hazards.id");
+        table.foreign("creator_role_id").references("roles.id");
+        table.foreign("actor_role_id").references("roles.id");
```

Comment:
@datajohnson Do these columns even make sense?
Why would the "role" of the actor and creator be relevant for linage?
I would expect these to be `creator_id` and `actor_id` linked to the `users.id` table.

Reply:
You are right... these names are inconsistent and IMO wrong. I brought these up in my meetin this morning and they are used that way in a couple places. Peter was thinking he would store which role took the action rather than just the specific user. I can see where he's coming from in a way, cause then you can say: "The Department Safety Officer approve this request" or whatever rather than simply : "Michael Johnson approved this request". Cause roles could be fluid over time. I will think more on this.

Reply:
Ah, so there are two kinds of roles then? e.g. "role" and "position"? Or are these concepts merged?